internal/core/bootstrap.go

```diff
@@ -36,57 +38,66 @@ type BootstrapConfig struct {
 // BootstrapConfig.Args is usually os.Args
 // BootstrapConfig.Commands is a list of command available in CLI.
 func Bootstrap(config *BootstrapConfig) (exitCode int, result interface{}, err error) {
+	// The global printer must be the first thing set in order to print errors
 	globalPrinter, err := printer.New(printer.Human, config.Stdout, config.Stderr)
 	if err != nil {
 		_, _ = fmt.Fprintln(os.Stderr, err)
 		return 1, nil, err
 	}
 
 	// Meta store globally available variables like SDK client.
-	// Meta is injected in a context object that will be pass to all commands.
-	m := &meta{
-		Printer:   globalPrinter,
+	// Meta is injected in a context object that will be passed to all commands.
+	meta := &meta{
 		BuildInfo: config.BuildInfo,
-		Client:    config.Client,
 		stdout:    config.Stdout,
 		stderr:    config.Stderr,
+		Client:    config.Client,
+		Commands:  config.Commands,
+		Printer:   globalPrinter,
+		result:    nil, // result is later injected by cobra_utils.go/cobraRun()
 	}
-	ctx := injectMeta(context.Background(), m)
-	ctx = injectCommands(ctx, config.Commands)
 
-	// Allocate space in ctx for command result
-	// result is later injected by cobra_utils.go/cobraRun()
-	ctx = injectResultSetter(ctx, &result)
+	// Send Matomo report when exiting the bootstrap
+	if (matomo.ForceTracking || config.BuildInfo.isRelease()) && matomo.IsTelemetryEnabled() {
+		start := time.Now()
+		defer func() {
+			matomoErr := matomo.TrackCommand(&matomo.TrackCommandRequest{
+				RunCommand:    meta.runCommand.getPath(),
```

Comment:
`RanCommand`

Reply:
This refers to the command that ran in the `Run` function ==> `Run command`

Reply:
why not just `Command` ? 

Can we have commands that are just typed, but noyt ran ?